tests/test_dicts.py

```diff
@@ -22,3 +22,21 @@ class MyClassFactory(ModelFactory[MyClass]):
     )
 
     assert obj.dict() == {"my_mapping_obj": {"baz": {"val": "bar"}}, "my_mapping_str": {"foo": "bar"}}
+
+
+def test_dict_with_union_random_types() -> None:
```

Comment:
So, please test using a random seed. Instead of checking after running 20 times, you should ensure deterministic results.

Reply:
The change was uploaded - lemme know if you mean sth like what I've pushed.